### PR TITLE
separate steps for Java and cpp codegen tests

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

**Comment by cseed** [comment]
Timing:

```
15543 | test_hail_java             | Complete | Success 🎉 (0) | 13 minutes | log
15544 | test_hail_java_cpp_codegen | Complete | Success 🎉 (0) | 12 minutes | log
```

File: build.yaml [hunk 1]
```diff
@@ -219,19 +219,40 @@ steps:
...
+ - kind: runImage
+   name: test_hail_java_cpp_codegen
+   image:
+     valueFrom: hail_run_image.image
+   script: |
+     set -ex
+     cd /io
+     mkdir -p src/test
+     tar xzvf resources.tar.gz -C src/test
+     export HAIL_TEST_SKIP_R=1
```
Comment (tpoterba) [inline build.yaml]: why mix export and inline variable assignment? maybe choose 1?

**Comment by tpoterba** [comment]
in other news, our tests have gotten slower again? I thought they were at about 15m...

**Comment by tpoterba** [comment]
Approving since the semantics didn't change, but we're not running the R test?

**Pull request merged**